modules/assignment/src/main/java/com/intuit/wasabi/assignment/cache/impl/AssignmentsMetadataCacheImpl.java

```diff
@@ -289,7 +259,14 @@ public BucketList getBucketList(Experiment.ID expId) {
             bucketList = (BucketList) val.getObjectValue();
         }
 
-        return isNull(bucketList)?new BucketList():bucketList;
+        return isNull(bucketList)?new BucketList():makeCopy(bucketList);
+    }
+
+    private BucketList makeCopy(BucketList bucketList) {
+        List<Bucket> iBucketList = new ArrayList<>(bucketList.getBuckets());
```

Comment:
@bargenilesh why is there ConcurrentModificationExcpetion for  BucketList? isn't the time interval 5 mins? how come multiple threads are trying to modify bucket list?